sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -591,20 +590,41 @@ case class FileSourceScanExec(
     logInfo(s"Planning scan with bin packing, max size: $maxSplitBytes bytes, " +
       s"open cost is considered as scanning $openCostInBytes bytes.")
 
+    // Filter files with bucket pruning if possible
```

Comment:
It's a bit odd that we call method name as `createNonBucketedReadRDD` but do something with buckets. I guess we could name `createNonBucketedReadRDD` like just `createReadRDD` or `createStandardReadRDD`